airflow/www/templates/airflow/graph.html

```diff
@@ -169,10 +169,27 @@
                                         "scale(" + d3.event.scale + ")");
           });
       svg.call(zoom);
-    }
 
-    // Centering the DAG on load (Not implemented yet)
-    // https://github.com/dagrejs/dagre-d3/issues/245
+      // Centering the DAG on load
+      // Get Dagre Graph dimensions
+      var graphWidth = g.graph().width;
+      var graphHeight = g.graph().height;
+      // Get SVG dimensions
+      var padding = 20;
+      var svgBb = svg.node().getBoundingClientRect();
+      var width = svgBb.width - padding*2;
+      var height = svgBb.height - padding;  // we are not centering the dag vertically
```

Comment:
What if some has set the orientation (default, or via drop down) to top->bottom? :)

Reply:
this calculation is orientation agnostics, it will setup the zoom level so all the dag nodes can fit into the current canvas regardless of the orientation.

Reply:
Oh it will fit the dag in vertically, just not centre it vertically. Gotcha.